sklearn/base.py

```diff
@@ -133,7 +142,121 @@ def _pprint(params, offset=0, printer=repr):
     return lines
 
 
-class BaseEstimator:
+class _PropsRequest:
+    def get_props_request(self):
+        """Get requested data properties.
+
+        Returns
+        -------
+        props : dict of list of strings, or dict of dict of {str: str}, or None
+            The key to the top level dict is the method for which the prop is
+            used. Under each key, there is a dict list of required properties,
+            or a dict of mapping of the form
+            ``{method_parameter: provided_parameter}``, or None.
+        """
+        try:
+            return self._props_request
+        except AttributeError:
+            return {}
+
+    def set_props_request(self, props):
```

Comment:
I think we should have this in a separate `PropConsumer` mixin. We only want it exposed on consumers that can have props provided. Ideally, too, each would validate the props provided (either at `fit` time or at `set_props_request` time) to ensure all will actually be consumed.

Reply:
Most meta estimators are _not_ prop consumers, only routers.